src/dodal/beamlines/i22.py

```diff
@@ -47,237 +48,141 @@
-        fake_with_ophyd_sim,
-        drv_suffix="CAM:",
-        hdf_suffix="HDF5:",
-        metadata_holder=NXSasMetadataHolder(
-            x_pixel_size=(1.72e-1, "mm"),
-            y_pixel_size=(1.72e-1, "mm"),
-            description="Dectris Pilatus3 2M",
-            type="Photon Counting Hybrid Pixel",
-            sensor_material="silicon",
-            sensor_thickness=(0.45, "mm"),
-            distance=(175.4199417092314, "mm"),
-        ),
+@device_factory()
+def waxs() -> PilatusDetector:
+    metadata_holder = NXSasMetadataHolder(
+        x_pixel_size=(1.72e-1, "mm"),
+        y_pixel_size=(1.72e-1, "mm"),
+        description="Dectris Pilatus3 2M",
+        type="Photon Counting Hybrid Pixel",
+        sensor_material="silicon",
+        sensor_thickness=(0.45, "mm"),
+        distance=(175.4199417092314, "mm"),
+    )
+    return NXSasPilatus(
+        prefix=f"{PREFIX.beamline_prefix}-EA-PILAT-03:",
         path_provider=get_path_provider(),
+        drv_suffix="CAM:",
+        hdf_suffix=HDF5_PREFIX,
+        metadata_holder=metadata_holder,
     )
 
 
-def i0(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> TetrammDetector:
-    return device_instantiation(
-        TetrammDetector,
-        "i0",
-        "-EA-XBPM-02:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-        type="Cividec Diamond XBPM",
+@device_factory()
+def i0() -> TetrammDetector:
+    return TetrammDetector(
+        prefix=f"{PREFIX.beamline_prefix}-EA-XBPM-02:",
         path_provider=get_path_provider(),
+        type="Cividec Diamond XBPM",
     )
 
 
-def it(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> TetrammDetector:
-    return device_instantiation(
-        TetrammDetector,
-        "it",
-        "-EA-TTRM-02:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-        type="PIN Diode",
+@device_factory()
+def it() -> TetrammDetector:
+    return TetrammDetector(
+        prefix=f"{PREFIX.beamline_prefix}-EA-TTRM-02:",
         path_provider=get_path_provider(),
+        type="PIN Diode",
     )
 
 
-def vfm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> FocusingMirror:
-    return device_instantiation(
-        FocusingMirror,
-        "vfm",
-        "-OP-KBM-01:VFM:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory()
+def vfm() -> FocusingMirror:
+    return FocusingMirror(
+        prefix=f"{PREFIX.beamline_prefix}-OP-KBM-01:VFM:",
     )
 
 
-def hfm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> FocusingMirror:
-    return device_instantiation(
-        FocusingMirror,
-        "hfm",
-        "-OP-KBM-01:HFM:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory()
+def hfm() -> FocusingMirror:
+    return FocusingMirror(
+        prefix=f"{PREFIX.beamline_prefix}|-OP-KBM-01:HFM:",
```

Comment:
Must: Accidental `|` after the prefix here.
```suggestion
        prefix=f"{PREFIX.beamline_prefix}-OP-KBM-01:HFM:",
```